tests/unit/models/PinsTest.php

```diff
@@ -49,4 +51,51 @@ public function testTogglePin(): void
         $this->assertTrue($this->Templates->Pins->isPinned());
         $this->assertTrue(count($this->Templates->Pins->readAll()) > 0);
     }
+
+    /**
+     * ensure that duplicated entry is not pinned
+     * @return void
+     */
+    public function testDuplicateIsNotPinned(): void
+    {
+        // duplicate existing experiment
+        $source = $this->Experiments;
+        $duplicate = $source->duplicate();
+        $freshSource = new Experiments($this->Users, $duplicate);
+        $this->assertNotEquals($source->entityData['id'], $freshSource->entityData['id']);
+
+        // ensure the duplicate experiment is not pinned
+        $this->assertCount(0, $freshSource->Pins->readAll());
+        $this->assertCount(0, $freshSource->Pins->readAllSimple());
+
+        // duplicate existing item
```

Comment:
in this case, you can factor it into a function that receives an Experiments or Items object. Here it's the exact same code twice, so a good candidate for factorisation into a function.